tests/utils/test_shell_util.py

```diff
@@ -426,49 +419,54 @@ def invoke(command):
                         child_processes.append(int(pids[0]))
                         parent_processes.append(int(pids[1]))
 
-                # the first item to in the PIDs we fetched corresponds run_get_output, which invokes the command using the
-                # shell, so in that case we need to use the parent's pid (which was the process we started)
-                started_commands_expected = parent_processes[0:1] + child_processes[1:]
+                # the first item to in the PIDs we fetched corresponds to run_get_output, which invokes the command using the
+                # shell, so in that case we need to use the parent's pid (i.e. the shell that we started)
+                started_commands = parent_processes[0:1] + child_processes[1:]
+
+                # wait for all the commands to start
+                running_commands = [[]]
+
+                def all_commands_running():
+                    running_commands[0] = shellutil.get_running_commands()
+                    return len(running_commands[0]) >= len(commands_to_execute) + 1  # +1 because run_pipe starts 2 commands
 
-                # wait for the command_started callbacks and verify them
-                if not self._wait_for(lambda: len(started_commands) == len(commands_to_execute) + 1):   # +1 because run_pipe starts 2 commands
-                    raise Exception("The child processes did not complete within the allowed timeout")
+                if not self._wait_for(all_commands_running):
+                    self.fail("shellutil.get_running_commands() did not report the expected number of commands after the allowed timeout. Got: {0}".format(
```

Comment:
Can we add the expected number to the error message?

Reply:
i'll report the expected commands

Reply:
done